File: scala/src/Problem2.scala
object Problem2 {
  def main(args : Array[String]) : Unit = {
    
    var sum = 0
    var i = 1
    var j = 2
    
    while (j < 4000000) {
      if (j % 2 == 0) 
        sum += j 
      val t = j
      j = i + j
      i = t
    }
    
    println("Solution: " + sum)
    
  }
}

File: java16/app/src/main/java/com/roblg/euler/soln/Problem2.java
package com.roblg.euler.soln;

import com.roblg.euler.util.FibonacciIterator;

public class Problem2 implements Solution {

    @Override
    public long calculate() {
        FibonacciIterator it = FibonacciIterator.create();

        long sum = 0;
        while (true) {
            long c = it.next();
            if (c > 4e6) {
                break;
            }
            if (c % 2 == 0) {
                sum += c;
            }
        }
        return sum;
    }
}
